src/webview_provider/sqlLineagePanel.ts

```diff
@@ -0,0 +1,318 @@
+      modelName,
+    );
+    if (!compiledSQL) {
+      return { errorMessage: "Unable to compile sql" };
+    }
+    const currNode = nodeMetaMap.get(modelName);
+    if (!currNode) {
+      return { errorMessage: "Unable to find model" };
+    }
+    const modelId = currNode.uniqueId;
+    const parentModels = graphMetaMap.parents.get(modelId)?.nodes || [];
+    const modelsToFetch = parentModels.map((n) => n.key);
+    // const { mappedNode } = await project.getNodesWithDBColumns(
+    //   event,
+    //   modelsToFetch,
+    //   token,
+    // );
+    const hash = crypto.createHash("md5").update(compiledSQL).digest("hex");
+    const sessionId = `${env.sessionId}-${hash}`;
+    const response = await this.altimate.sqlLineage({
+      compiled_sql: compiledSQL,
+      // model_info: modelsToFetch.map((n) => ({ model_node: mappedNode[n] })),
+      model_info: [],
+      model_dialect: project.getAdapterType(),
+      session_id: sessionId,
+    });
+    const { details } = response;
+
+    const nodeTypeMapping: Record<string, string> = {};
+    for (const modelId of modelsToFetch) {
+      const splits = modelId.split(".");
+      if (splits[0] === "source") {
+        const _source = sourceMetaMap.get(splits[splits.length - 2]);
+        const _table = splits[splits.length - 1].toLowerCase();
+        if (_source) {
+          for (const key in details) {
+            if (details[key].type === "table" && key.toLowerCase() === _table) {
+              nodeTypeMapping[key] = "source";
+              break;
+            }
+          }
+          continue;
+        }
+      }
+      const _node = nodeMetaMap.get(splits[splits.length - 1]);
+      if (_node) {
+        for (const key in details) {
+          if (
+            details[key].type === "table" &&
+            key.toLowerCase() === _node.alias.toLowerCase()
+          ) {
+            nodeTypeMapping[key] = _node.resource_type;
+            break;
+          }
+        }
+        continue;
+      }
+    }
+    nodeTypeMapping[modelName] = currNode.resource_type;
+    const FINAL_SELECT = "__final_select__";
+    const tableEdges = response.tableEdges.map(
+      (edge) =>
+        edge.map((item) => (item === FINAL_SELECT ? modelName : item)) as [
+          string,
+          string,
+        ],
+    );
+    details[modelName] = details[FINAL_SELECT];
+    delete details[FINAL_SELECT];
+    for (const k in details) {
+      details[k]["nodeType"] = nodeTypeMapping[k];
+      if (k === modelName) {
+        details[k]["name"] = modelName;
+      }
+    }
+    return { tableEdges, details };
+  }
+
+  resolveWebviewView(
+    panel: WebviewPanel,
+    lineage: SQLLineage,
+  ): void | Thenable<void> {
+    this._panel = panel;
+    this.terminal.debug(
+      "sqlLineagePanel:resolveWebviewView",
+      "onResolveWebviewView",
+    );
+    this.setupWebviewOptions();
+    this.renderWebviewView();
+    this.changedActiveColorTheme();
+    this._panel?.webview.postMessage({
```

Comment:
does the webview receive this message? because the webview will not complete the loading by this time and this message could be lost